doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,181 @@
+complexity to support fully customizable artist styline and statistics
+computation. This lead to ``Axes.boxplot`` being split off into multiple
+parts. The statistics needed to draw a boxplot are computed in
+``cbook.boxplot_stats``, while the actual artists are drawn by ``Axes.bxp``.
+The original method, ``Axes.boxplot`` remains as the most public API that
+handles passing the user-supplied data to ``cbook.boxplot_stats``, feeding
+the results to ``Axes.bxp``, and pre-processing style information for
+each facet of the boxplot plots.
+
+This MEP will outline a path forward to roll-back the added complexity
+and simplify the API while maintaining reasonable backwards
+compatibility.
+
+Detailed description
+====================
+
+Currently, the ``Axes.boxplot`` method accepts parameters that allow the
+users to specify medians and confidence intervals for each box that
+will be drawn in the plot. These were provided so that avdanced users
+could provide statistics computed in a different fashion that the simple
+method provided by matplotlob. However, handling this input requires
+complex logic to make sure that the forms of the data structure match what
+needs to be drawn. At the moment, that logic contain 9 separate if/else
+statements nested up to 5 levels deep with a for loop, and raise up to 2 errors.
+These parameters were added prior to the creation of the ``Axes.bxp`` method,
+which draws boxplots from a list of dictionaries containing the relevant
+statistics. Matplotlib also provides a function that computes these
+statistics via ``cbook.boxplot_stats``. Note that advanced users can now
+either a) write their own function to compute the stats required by
+``Axes.bxp``, or b) modify the output returned by ``cbook.boxplots_stats``
+to fully customize the position of the artists of the plots. With this
+flexibility, the parameters to manually specify only the medians and their
+confidences intervals remain for backwards compatibility.
+
+Around the same time that the two roles of ``Axes.bxplot`` were split into
+``cbook.boxplot_stats`` for computation and ``Axes.bxp`` for drawing, both
+``Axes.boxplot`` and ``Axes.bxp`` were written to accept parameters that
+individually toggle the drawing of all components of the boxplots, and
+parameters that individually configure the style of those artists. However,
+to maintain backwards compatibility, the ``sym`` parameter (previously used
+to specify the symbol of the fliers) was retained. This parameter itself
+requires fairly complex logic to reconcile the ``sym`` parameters with the
+newer ``flierprops`` parameter at the default style specified by ``matplotlibrc``.
+
+This MEP seeks to dramatically simplify the creation of boxplots for
+novice and advanced users alike. Importantly, the changes proposed here
+will also be available to downstream packages like seaborn, as seaborn
+smartly allows users to pass arbitrary dictionaries of parameters through
```

Comment:
you're absolutely right about that. i guess it just makes me feel insecure about our implementation.